core/src/main/scala/org/apache/spark/rdd/InputFileBlockHolder.scala

```diff
@@ -76,7 +76,6 @@ private[spark] object InputFileBlockHolder {
   def set(filePath: String, startOffset: Long, length: Long): Unit = {
     require(filePath != null, "filePath cannot be null")
     require(startOffset >= 0, s"startOffset ($startOffset) cannot be negative")
-    require(length >= 0, s"length ($length) cannot be negative")
```

Comment:
If `-1` is valid, why not `length >= -1` instead of `length >= 0`?

Reply:
Hi @dongjoon-hyun 

The file InputFileBlockHolder contains the following statement stating that -1 can be returned as length if the length is unknown:
```
  /**
   * Returns the length of the block being read, or -1 if it is unknown.
   */
  def getLength: Long = inputBlock.get().get().length
```

Yes, I can modify the validation statement to be `length >= -1`. In that case, what can be a good message? Maybe - "length ($length) cannot be smaller than -1"?

Reply:
First of all, it's `private` to Spark. You should not depend on the private implementation of Apache Spark. There is no guarantee for that and this is not a regression of Apache Spark.

Reply:
Yes. Please update the PR with `length >= -1`. For the message, that message looks okay. Then, probably, it's okay.
> Yes, I can modify the validation statement to be length >= -1. In that case, what can be a good message? Maybe - "length ($length) cannot be smaller than -1"?